src/EnergyPlus/HeatBalanceSurfaceManager.cc

```diff
@@ -6704,6 +6720,705 @@ namespace HeatBalanceSurfaceManager {
+
+        int OtherSideSurfNum;     // Surface number index for other side of an interzone partition
+        static int MinIterations; // Minimum number of iterations for the inside heat balance
+        static int ErrCount(0);
+        Real64 Ueff; // 1 / effective R value between TDD:DOME and TDD:DIFFUSER
+
+        int ZoneEquipConfigNum;
+        int NodeNum;
+        Real64 SumSysMCp;  // Zone sum of air system MassFlowRate*Cp
+        Real64 SumSysMCpT; // Zone sum of air system MassFlowRate*Cp*T
+        Real64 MassFlowRate;
+        Real64 NodeTemp;
+        Real64 CpAir;
+        static Array1D<Real64> RefAirTemp; // reference air temperatures
+        static bool MyEnvrnFlag(true);
+        static int InsideSurfErrCount(0);
+        static int WarmupSurfTemp;
+        static int TimeStepInDay(0); // time step number
+
+        if (calcHeatBalanceInsideSurfCTFOnlyFirstTime) {
+            TempInsOld.allocate(TotSurfaces);
+            RefAirTemp.allocate(TotSurfaces);
+            MinIterations = 1;
+            if (DisplayAdvancedReportVariables) {
+                SetupOutputVariable("Surface Inside Face Heat Balance Calculation Iteration Count",
+                                    OutputProcessor::Unit::None,
+                                    InsideSurfIterations,
+                                    "ZONE",
+                                    "Sum",
+                                    "Simulation");
+            }
+            // Precompute whether CTF temperature limits will be needed
+            DataHeatBalSurface::Zone_has_mixed_HT_models.resize(NumOfZones + 1, false);
+            for (int iZone = 1; iZone <= NumOfZones; ++iZone) {
+                auto const &zone(Zone(iZone));
+                for (int iSurf = zone.SurfaceFirst, eSurf = zone.SurfaceLast; iSurf <= eSurf; ++iSurf) {
+                    auto const alg(Surface(iSurf).HeatTransferAlgorithm);
+                    if ((alg == HeatTransferModel_CondFD) || (alg == HeatTransferModel_HAMT) || (alg == HeatTransferModel_Kiva)) {
+                        DataHeatBalSurface::Zone_has_mixed_HT_models[iZone] = true;
+                        break;
+                    }
+                }
+            }
+            // Set up coefficient arrays that never change - loop over non-window HT surfaces
+            for (int zoneNum = FirstZone; zoneNum <= LastZone; ++zoneNum) {
+                int const firstSurf = Zone(zoneNum).NonWindowSurfaceFirst;
+                int const lastSurf = Zone(zoneNum).NonWindowSurfaceLast;
+                for (int surfNum = firstSurf; surfNum <= lastSurf; ++surfNum) {
+                    int const ConstrNum = Surface(surfNum).Construction;
+                    auto const &construct(Construct(ConstrNum));
+                    if (Surface(surfNum).ExtBoundCond == surfNum) {
+                        IsAdiabatic(surfNum) = 1;
+                        IsNotAdiabatic(surfNum) = 0;
+                    } else {
```

Comment:
The original `CalcHeatBalanceInsideSurf2` had a long set of nested if blocks with similar but different equations for various combinations of surface properties (adiabatic, interzone, swimming pool, internal source, etc.). This is all collapsed below into a few general equations that have all possible terms. These integer 1/0 variables are used to include/exclude applicable terms. Not sure if this is the best long-term solution, but it helped facilitate auto-vectorization for those calculations. Most of these are fixed for the entire simulation, but `IsPool` changes based on whether there is source energy to the pool, so that gets re-checked later on.